### DISCUSSION TITLE
Is there a way to get destination direcory in template?

### DISCUSSION THREAD

**Discussion by orzomaxx** [twpayne/chezmoi#3599] [Q&A] [answered]
Hi,
I had this line for some time in my `.chezmoi.yaml.tmpl` to save `sourceDir` from the first call to `chezmoi init`: 
```
sourceDir: {{ .chezmoi.sourceDir }}
```

Today I tried to do the same to `destDir` and failed. Seems that there is `destDir: ...` in config but no `.chezmoi.destDir` in variables.

Is there a way to get `destDir`, other than parsing `.chezmoi.args` for `--destination` argument?

**Comment by halostatue** [comment]
You can definitely get the value from `.chezmoi.config.destDir`, but *most of the time* that will be the same as `.chezmoi.homeDir`.

It feels like there should be a `.chezmoi.destDir` and it should be preferred over `.chezmoi.homeDir` inside of templates so that things which are supposed to be filled in based on what will be resolved (or what has been resolved) can be resolved to the correct path. If I had `~/.config` as a symlink to `~/dotfiles/config` and set `destDir = "~/dotfiles"`, then I should be adding paths based on `~/dotfiles` as my root rather than `~`.

**Comment by orzomaxx** [comment, marked as answer]
Managed to implement one-time save of `destDir`:

```
destDir: &destDir {{ dig "cz" "destDir" .chezmoi.config.destDir . }}

data:
  cz:
    destDir: *destDir
```
It is trying to use saved `destDir`, or if this is first `init` it uses current `destDir` (possibly provided via `--destination` argument) and saves it.

Thanks @halostatue for pointing me to `.chezmoi.config.destDir`!

UPD: Changed usage of `.chezmoi` to `.cz` dict

**Reply by halostatue** [reply]
I *strongly* recommend *against* modifying `.chezmoi` values in your configuration file. This is not supported and is likely working by accident.

**Reply by orzomaxx** [reply]
Thanks for advice, moved everyting into `.cz` dict
